client/peerset/src/lib.rs

```diff
@@ -714,52 +712,27 @@ impl Peerset {
 	}
 
 	/// Returns the number of peers that we have discovered.
-	pub fn num_discovered_peers(&self) -> usize {
+	fn num_discovered_peers(&self) -> usize {
 		self.data.peers().len()
 	}
 }
 
 impl Stream for Peerset {
 	type Item = Message;
 
-	fn poll_next(mut self: Pin<&mut Self>, cx: &mut Context) -> Poll<Option<Self::Item>> {
-		loop {
-			if let Some(message) = self.message_queue.pop_front() {
-				return Poll::Ready(Some(message))
-			}
-
-			if Future::poll(Pin::new(&mut self.next_periodic_alloc_slots), cx).is_ready() {
-				self.next_periodic_alloc_slots = Delay::new(Duration::new(1, 0));
-
-				for set_index in 0..self.data.num_sets() {
-					self.alloc_slots(SetId(set_index));
-				}
-			}
-
-			let action = match Stream::poll_next(Pin::new(&mut self.rx), cx) {
-				Poll::Pending => return Poll::Pending,
-				Poll::Ready(Some(event)) => event,
-				Poll::Ready(None) => return Poll::Pending,
-			};
+	fn poll_next(self: Pin<&mut Self>, cx: &mut Context) -> Poll<Option<Self::Item>> {
```

Comment:
I'm not a huge fan of this design choice. Previously there was a clear difference between the protocols that would use `Peerset` and the protocol that owned `Peerset` and I think it made sense. Now it seems that any protocol that acquires a handle to `Peerset` can poll it and drain events from it even if it has no idea what to do with them.

Reply:
Would splitting `Peerset` into `Peerset` (lacking polling) and `PeersetStream`, both actually containing an `Arc` make it better?

Reply:
The naming can be different, of course.

Reply:
I think splitting them into two would be good.